workaround apt source issue of debian8

amoskong opened the pull request:
jessie-updates and jessie-backports have been removed from debian 8 source,
apt-get update/upgrade will fail. This patch fixed the source path and name.

Effected:
- scylla installation
- install scylla manager
- loader setup

amoskong commented:
This PR had been used in debian8 upgrade testing, apt source issue doesn't happen.

http://jenkins.cloudius-systems.com:8080/job/scylla-3.0/job/3.0-test/job/scylla-rolling-upgrade-debian8-test/

roydahan commented:
Is this in addition to PR from @yarongilor?
Instead of them?
Can someone please explain the issue where we have it and where not?
If it's only for deb8, let's drop support for it.
Do we need it only in manager?

amoskong commented:
On Sun, Apr 21, 2019 at 5:42 PM Roy Dahan <notifications@github.com> wrote:

> Is this in addition to PR from @yarongilor <https://github.com/yarongilor>
> ?
>
Yaron's PR scylla-cluster-tests/pull/876 had been reviewed many days. It
still can't solve my problem in rolling upgrade / all gce backend testing,
some blocked me. This PR corrected something & add more fix.

There are also some bug in dedent string of install_java_script of debian
8, which is fixed in this PR.

> Instead of them?
>
Yaron's patch has some fix of scylla manager, like sleep 5 seconds. this PR
can't instead of them.
We can review both PR together, apply right one (addressed all comments),
and rebase another one.

Can someone please explain the issue where we have it and where not?
>
ANY debian 8 machine will has problem in `apt-get update`, if the systems
use original default source file (which isn't available now)
- debian 8 that is used as db node/loader/monitor
- aws/gce/etc
- debian 8 VM that is used for artifact-test

> If it's only for deb8, let's drop support for it.
>
 will we still test rolling upgrade of latest scylla 2.3 / 3.0 with debian
8? just drop it?

> Do we need it only in manager?
>
ALL debian 8 system, everywhere.

amoskong commented:
On Sun, Apr 21, 2019 at 11:24 PM Amos Kong <amos@scylladb.com> wrote:

> On Sun, Apr 21, 2019 at 5:42 PM Roy Dahan <notifications@github.com>
> wrote:
>
>> Is this in addition to PR from @yarongilor
>> <https://github.com/yarongilor>?
>>
> Yaron's PR scylla-cluster-tests/pull/876 had been reviewed many days. It
> still can't solve my problem in rolling upgrade / all gce backend testing,
> some blocked me. This PR corrected something & add more fix.
>
> There are also some bug in dedent string of install_java_script of debian
> 8, which is fixed in this PR.
>
>> Instead of them?
>>
> Yaron's patch has some fix of scylla manager, like sleep 5 seconds. this
> PR can't instead of them.
> We can review both PR together, apply right one (addressed all comments),
> and rebase another one.
>
> Can someone please explain the issue where we have it and where not?
>>
> ANY debian 8 machine will has problem in `apt-get update`, if the systems
> use original default source file (which isn't available now)
> - debian 8 that is used as db node/loader/monitor
> - aws/gce/etc
> - debian 8 VM that is used for artifact-test
>
>> If it's only for deb8, let's drop support for it.
>>
>  will we still test rolling upgrade of latest scylla 2.3 / 3.0 with debian
> 8? just drop it?
>

Same question for the artifact-test with debian8 of Scylla latest 2.3 / 3.0?

Do we need it only in manager?
>>
> ALL debian 8 system, everywhere.
>

amoskong commented:
ping @roydahan

bentsi commented:
@roydahan in #876 you said that we will drop support for Debian 8. If so should we close this PR?

roydahan commented:
let's merge for now.
Not dropping all deb8 yet.

bentsi commented:
do we need to backport this to some other branches?

roydahan commented:
@amoskong, labels?

bentsi commented:
2019.1 - backported
3.0 - backported all except changes in `sdcm.cluster.BaseNode#install_mgmt` since they are manager related and we run manager on separate branches
